ast/compile_test.go

```diff
@@ -109,6 +110,122 @@ func TestCompilerExample(t *testing.T) {
 	assertNotFailed(t, c)
 }
 
+func TestCompilerUserFunctionConflicts(t *testing.T) {
+	tests := []struct {
+		module string
+		assert func(*testing.T, *Compiler)
+	}{
+		{
+			module: `package x
+
+				f([x]) = y {
+					y = x
+				}
+
+				f({"foo": x}) = y {
+					y = x
+				}`,
+			assert: assertFailed,
```

Comment:
I reviewed a few more failure cases below. I don't understand why they are marked as failures. Notes or error messages would be really helpful.